qemu/tests/chardev_free_port.py

```diff
@@ -0,0 +1,46 @@
+from virttest import error_context, env_process
+from qemu.tests.virtio_console import add_chardev
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    qemu should try to find a free port value by to=<port> with unix socket and tcp options:
+    1) boot guest with socket with -qmp tcp:0:4444.
+    2) telnet localhost.
+    3) check output of {"execute": "qmp_capabilities"}, {"execute":"query-chardev"}.
+    4) repeat step 1 to 3 with -qmp tcp:0:4445 and telnet to port 4445.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    chardev_infos = []
+    port_nums = []
+    vms = ['vm1', 'vm2']
+    for ind, vm in enumerate(vms):
+        vm1 = env.get_vm(vm)
+        char_devices = add_chardev(vm1, params)
```

Comment:
For the second vm, it will get a new port, which is different with the first one. But in our test, they must be same.